src/defs.js

```diff
@@ -98,6 +99,15 @@ class CustomDef extends Def {
         return typeDef && typeDef.spec && typeDef.spec.pii || false;
     }
 
+    isCanonical() {
+        return this.spec.canonical || false;
+    }
+
+    getCanonicalFields() {
+        const typeDef = this.getTypeDef();
+        return getCanonicalFields(typeDef) || null;
```

Comment:
It's hard to reason about the returned data type here (is it `Array<string> | null`?) Would it be better to guarantee that it always returns an array?

Also, how would client library use that? I reckon it would need to obtain the values of canonical fields, based on this meta information. I wonder it it's easier to just list all json pointers to canonical fields directly on `InputDef` (and therefore leave json schema part of protocol unmodified)

Reply:
Sorry, scratch this "json pointer" things above. I reckon it won't work for passengers...